api/v1alpha1/companion_types.go

```diff
@@ -17,37 +17,105 @@ limitations under the License.
 package v1alpha1
 
 import (
+	kcorev1 "k8s.io/api/core/v1"
 	kmetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-
 const (
 	StateReady      string = "Ready"
 	StateError      string = "Error"
 	StateProcessing string = "Processing"
+	StateDeleting   string = "Deleting"
 	StateWarning    string = "Warning"
 )
 
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // CompanionSpec defines the desired state of Companion.
 type CompanionSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+
+	// Companion configuration.
+	// +kubebuilder:default:={namespace:"companion",deploymentNamespace:"ai-core",configMapNames:{"companion-config"},secretsNames:{"companion-secrets"},replicas:1}
+	CompanionConfig CompanionConfig `json:"companionConfig"`
+
+	// HANA Cloud configuration.
+	// +kubebuilder:default:={namespace:"hana-cloud"}
+	HanaCloudConfig HanaCloudConfig `json:"hanaCloudConfig"`
+
+	// Redis configuration.
+	// +kubebuilder:default:={namespace:"redis"}
+	RedisConfig RedisConfig `json:"redisConfig"`
+
+	// Container port for the companion backend. Default value is 5000.
+	// ContainerPort int32 `json:"containerPort"`
+
+	// Annotations allows to add annotations to NATS.
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Labels allows to add Labels to NATS.
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// CompanionConfig defines the configuration for the Companion.
+type CompanionConfig struct {
+	// Companion namespace where the companion backend will be deployed
+	// and the related configMaps and secrets are already stored.
+	// +kubebuilder:default:="ai-core"
+	Namespace string `json:"namespace"`
+	// ConfigMap names for the companion backend.
+	ConfigMapNames []string `json:"configMapNames"`
+	// Secret names for the companion backend.
+	SecretsNames []string `json:"secretsNames"`
+
+	// Specify required resources and resource limits for the companion backend.
+	// Example:
+	// resources:
+	//   limits:
+	//     cpu: 1
+	//     memory: 1Gi
+	//   requests:
+	//     cpu: 500m
+	//     memory: 256Mi
+	// +kubebuilder:default:={resources:{limits:{cpu:"4",memory:"4Gi"},requests:{cpu:"500m",memory:"256Mi"}}}
+	// +kubebuilder:default:={limits:{cpu:"4",memory:"4Gi"}, requests:{cpu:"500m",memory:"256Mi"}}
+	Resources kcorev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+// hanaCloudConfig defines the configuration for the HANA Cloud.
+type HanaCloudConfig struct {
+	// HANA Cloud namespace where the HANA Cloud is deployed.
+	// +kubebuilder:default:="hana-cloud"
+	Namespace string `json:"namespace"`
+}
+
+// redisConfig defines the configuration for the Redis.
```

Comment:
nit: Same as for HannaCloudConfig.

Reply:
Please focus on concept, and the structure. This is only an initial draft.